pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -552,21 +579,39 @@ func convertFromAzureSecurityRuleToInternalEgressRule(rule armnetwork.SecurityRu
 	if err != nil {
 		return nil, err
 	}
+	cloudResourceID := securitygroup.CloudResourceID{
```

Comment:
Instead of creating cloudResourceID here, the caller can create and pass the appliedToGroup name.